### ISSUE TITLE
Avatar component

### ISSUE THREAD

**Issue by mwcz** [patternfly/patternfly-elements#214]
- [x] use CSS variables for the color
- [x] hash name attribute
- [x] draw pattern to canvas based on hash number
- [x] display initials
- [x] decide on canvas vs svg
- [x] allow using a provided image instead of generated pattern
- [x] for displaying the provided image, decide on `<img>` vs creating a `<rh-img>`
- [x] allow square, rounded square, and circular display
- [x] decide which CSS vars to use, and how to customize them
- [x] come up with a set of 5 or 6 colors (I just used a few colors I saw at brand.redhat.com)
- [x] implement splitting of `--rh-avatar--colors` values (space delim) and using name hash to pick which color (with mod)
- [x] add HSL lib and darken selected color (or lighten, if the color is too close to black)
- [x] figure out sizing of the element and the text within (skipping this in favor of a broader solution related to #238)
- [x] add markup preview to storybook
- [x] write rh-avatar's README
- [x] determine how to choose initials (maybe 1, 2, or 3 instead of just 1 or 2)
- [x] use squares on the Portal 
- [x] write documentation

**Comment by mwcz** [comment]
ALMOST DONE!

**Comment by ajacobs21e** [comment]
https://www.w3.org/International/questions/qa-personal-names
tl;dr there are so many more cultural name variations than I ever knew and I don't think we can come up with a rule that suits all names.

Options:
* stick with Michael's latest algorithm (2 or 3 initials)
** won't cover every use case, but probably a majority
* no initials
** could introduce confusion / lack of recognizability between similar avatars
** simplest from a logic standpoint
* only one initial
** may be the simplest and still maintain recognizability of your avatar
* add option in the profile to either upload your own picture or add initials to your generated avatar and let the user pick 2-3 initials
** requires an extra step for the user to add initials but the user will pick ones that work for their name
** will users abuse this and enter inappropriate initials?

I am leaning towards no initials. I want to be highly sensitive to issues of personal identity and do not want to choose a format that is not inclusive. This is meant to be only a default avatar - users will always have the option to upload something personalized. Therefore the default avatar can be more basic.

@markcaron & @kylebuch8 Do you still support the first option (2-3 initials)?

**Comment by mwcz** [comment]
My vote is for no initials.  It keeps the element simple, avoiding a lot of complexity related to guess initials.  If the current patterns/colors don't offer enough variety, I can add more types of patterns and we can add as many color variations as we want.

**Comment by ajacobs21e** [comment]
@mwcz Is this ready to be finished now? Do you need any more input from me? Thanks!

**Comment by mwcz** [comment]
@ajacobs21e Yep!  The PR is finished and ready to be merged once it's code reviewed by someone.

**Comment by markcaron** [comment]
I still need to update the default colors.

**Comment by castastrophe** [comment]
@mwcz and I discussed possibly merging pfe-avatar and going back in to update the colors when @markcaron has time/ability to do so.  I think that's ideal since the change in colors wouldn't be a breaking change.

**Comment by castastrophe** [comment]
@ajacobs21e I don't see a reply to you on this thread so I'll do my best; the avatar will use images and automatically fallback to a randomly generated pattern if none is provided.

**Comment by ajacobs21e** [comment]
Thanks @castastrophe . My question was more about if users' uploaded photos will be available for other apps/sites to use (e.g. Insights, Developers, etc)

**Comment by castastrophe** [comment]
I think that would be a back-end wiring kind of thing not necessarily built into the component.  So we could all agree on a service or build an API that could server up the image and an app could request the image and then pass it into the component but the component itself wouldn't be doing that part of it because that could limit its usability across assets.

**Comment by ajacobs21e** [comment]
Thank you! That's exactly what I needed to know :)
